plugins/services/src/js/components/ServiceStatusWarning.js

```diff
@@ -7,27 +7,52 @@ import Icon from '../../../../../src/js/components/Icon';
 import Pod from '../structs/Pod';
 import Service from '../structs/Service';
 import ServiceTree from '../structs/ServiceTree';
+import StringUtil from '../../../../../src/js/utils/StringUtil';
 
 class ServiceStatusWarning extends Component {
-  getTooltipContent(timeWaiting) {
-    return `DC/OS has been waiting for resources and is unable to complete this deployment for ${DateUtil.getDuration(timeWaiting, null)}.`;
+  getTooltip(content) {
+    return (
+      <Tooltip
+        content={content}
+        width={250}
+        wrapText={true}
+        wrapperClassName="tooltip-wrapper status-waiting-indicator">
+        <Icon color="red" id="yield" size="mini" />
+      </Tooltip>
+    );
   }
+
+  getAppsWithDeclinedOffers(serviceTree) {
```

Comment:
I don't understand what you mean by this, are you saying I shouldn't add a `getAppsWithDeclinedOffers` method to this component? I think that's what you mean, so that's what I did here: https://github.com/dcos/dcos-ui/pull/1749/commits/94218f6b7b584bac22e987d1b21369622ebfb383

Could you expand more on the reasoning for this? To me the method call made the logic easier to follow.